scipy/spatial/transform/_rotation.pyx

```diff
@@ -2806,54 +2833,237 @@ cdef class Rotation:
+            raise ValueError("Expected input `a` to have shape (3,) or "
+                             "(N, 3), got {}".format(a.shape))
+        if b.shape[-1] != 3:
+            raise ValueError("Expected input `b` to have shape (3,) or "
+                             "(N, 3), got {}".format(b.shape))
         if a.shape != b.shape:
             raise ValueError("Expected inputs `a` and `b` to have same shapes"
                              ", got {} and {} respectively.".format(
-                                a.shape, b.shape))
+                                 a.shape, b.shape))
+        a = np.atleast_2d(a)
+        b = np.atleast_2d(b)
+        N = len(a)
 
+        # Check weights
         if weights is None:
-            weights = np.ones(len(b))
+            weights = np.ones(N)
         else:
-            weights = np.asarray(weights)
-            if weights.ndim != 1:
-                raise ValueError("Expected `weights` to be 1 dimensional, got "
-                                 "shape {}.".format(weights.shape))
-            if weights.shape[0] != b.shape[0]:
-                raise ValueError("Expected `weights` to have number of values "
-                                 "equal to number of input vectors, got "
-                                 "{} values and {} vectors.".format(
-                                    weights.shape[0], b.shape[0]))
-            if (weights < 0).any():
-                raise ValueError("`weights` may not contain negative values")
-
+            weights = np.array(weights, dtype=float)
+
+        if weights.ndim != 1:
+            raise ValueError("Expected `weights` to be 1 dimensional, got "
+                             "shape {}.".format(weights.shape))
+        if N > 1 and (weights.shape[0] != N):
+            raise ValueError("Expected `weights` to have number of values "
+                             "equal to number of input vectors, got "
+                             "{} values and {} vectors.".format(
+                                 weights.shape[0], N))
+        if (weights < 0).any():
+            raise ValueError("`weights` may not contain negative values")
+
+        # For the special case of a single vector pair, we use the infinite
+        # weight code path
+        if N == 1:
+            weight_is_inf = np.array([True])
+        else:
+            weight_is_inf = np.isposinf(weights)
+        n_inf = np.sum(weight_is_inf)
+
+        # Check for an infinite weight, which indicates that the corresponding
+        # vector pair is the primary unmoving reference to which we align the
+        # other secondary vectors
+        if n_inf > 1:
+            raise ValueError("Only one infinite weight is allowed")
+
+        elif n_inf == 1:
+            if return_sensitivity:
+                raise ValueError("Cannot return sensitivity matrix with an "
+                                 "infinite weight or one vector pair")
+            a_pri = a[weight_is_inf]
```

Comment:
Why not take `[0]` right here?

Also, you of course may adjust the names as you see fit, `a_pri` and `b_pri` seems fine though.

Reply:
einsum down below needs this to be a 2d input to work, and I feel like it's cleaner as-is rather than converting down below